pkg/kubelet/kuberuntime/kuberuntime_container_linux.go

```diff
@@ -100,7 +100,15 @@ func (m *kubeGenericRuntimeManager) generateLinuxContainerResources(pod *v1.Pod,
 	if _, cpuRequestExists := container.Resources.Requests[v1.ResourceCPU]; cpuRequestExists {
 		cpuRequest = container.Resources.Requests.Cpu()
 	}
-	lcr := m.calculateLinuxResources(cpuRequest, container.Resources.Limits.Cpu(), container.Resources.Limits.Memory())
+
+	// If pod has exclusive cpu and the container in question has integer cpu requests
+	// the cfs quota will not be enforced
+	enforceCfsQuota := true
+	if utilfeature.DefaultFeatureGate.Enabled(kubefeatures.DisableCPUQuotaWithExclusiveCPUs) && m.containerManager.HasExclusiveCPUs(pod) && container.Resources.Requests.Cpu().Value()*1000 == container.Resources.Requests.Cpu().MilliValue() {
```

Comment:
`HasExclusiveCPUs()` will return `true` if and only if the request value is integer, that's guaranteed and enforced by cpumanager, so we can remove that rendundant guard - and reduce the coupling between subsystems.